core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -702,30 +703,34 @@ public PlanWithProperties visitTableExecute(TableExecuteNode node, StreamPreferr
                     node.getPartitioningScheme(),
                     node.getSource(),
                     parentPreferences,
-                    node.getTarget());
+                    node.getTarget(),
+                    // Disable task writer scaling for TableExecute since it can lead to file sizes which are not
+                    // controlled via OPTIMIZE command thus makes it unreliable.
```

Comment:
`since it can lead to file sizes which are not controlled via OPTIMIZE command thus makes it unreliable` -> `since it can result in smaller files than file_size_threshold, which can be undesirable behaviour`